File: measure.js
const http = require('http');

let fetchCount = 0;
const start = Date.now();

const doFetch = () => {
  const req = http.get('http://localhost:8000/api/stats', (res) => {
    res.on('data', () => {});
    res.on('end', () => {
      fetchCount++;
      if (Date.now() - start < 10000) {
        setTimeout(doFetch, 2000);
      } else {
        console.log(`Made ${fetchCount} requests in 10 seconds.`);
      }
    });
  });
  req.on('error', (e) => {
    console.error(`Problem with request: ${e.message}`);
  });
};

// doFetch();

File: measure.py
import urllib.request
import time

start_time = time.time()
requests = 0
while time.time() - start_time < 5:
    urllib.request.urlopen("http://localhost:8000/api/stats").read()
    urllib.request.urlopen("http://localhost:8000/api/trades").read()
    requests += 2
    time.sleep(2)
print(f"Polling baseline: {requests} HTTP requests in 5s")
